engine/src/main/java/org/terasology/logic/console/commands/CoreCommands.java

```diff
@@ -461,12 +465,30 @@ public String leave() {
      * @throws IOException thrown when error with saving file occures
      */
     @Command(shortDescription = "Writes out information on all entities to a text file for debugging",
-            helpText = "Writes entity information out into a file named \"entityDump.txt\".")
-    public void dumpEntities() throws IOException {
+            helpText = "Writes entity information out into a file named \"entityDump.txt\"." +
+                    " Supports list of component names, which will be used to only save entities that contains" +
+                    " all or some of those components. List of component names must be provided in double quotes" +
+                    " and all names must be separated by spaces.")
+    public void dumpEntities(@CommandParam(value = "componentNames", required = false) String componentNames) throws IOException {
         EngineEntityManager engineEntityManager = (EngineEntityManager) entityManager;
         PrefabSerializer prefabSerializer = new PrefabSerializer(engineEntityManager.getComponentLibrary(), engineEntityManager.getTypeSerializerLibrary());
         WorldDumper worldDumper = new WorldDumper(engineEntityManager, prefabSerializer);
-        worldDumper.save(PathManager.getInstance().getHomePath().resolve("entityDump.txt"));
+        if (componentNames == null) {
+            worldDumper.save(PathManager.getInstance().getHomePath().resolve("entityDump.txt"));
+        } else {
+            Reflections reflections = new Reflections("org.terasology");
+            List<Class<? extends Component>> filterComponents = new LinkedList<>();
+            List<String> componentNamesList = Arrays.asList(componentNames.split(" "));
+            for (String componentName : componentNamesList) {
+                if (componentName.trim().length() > 0) {
+                    Optional<Class<? extends Component>> component = reflections.getSubTypesOf(Component.class).stream().filter(e -> e.getSimpleName().equals(componentName)).findFirst();
+                    component.ifPresent(filterComponents::add);
+                }
+            }
+            if (!filterComponents.isEmpty()) {
```

Comment:
Very much agreed, had me a little confused during testing as I picked a few  examples that actually didn't result in any entities, so I didn't get an updated dump file or if I had deleted it then I'd get no file at all 😁 

It would be sweet getting some return text in either case of how many matched entities were saved, but I'm not sure how readily available that info would be.

Reply:
Done.

Reply:
👍